src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java

```diff
@@ -305,6 +333,81 @@ private String postTestWebResource() throws Exception {
         return mvcResult.getResponse().getHeader("Location");
     }
 
+    private String postTestHTTPWebResource() throws Exception {
```

Comment:
You can group these four "repeated" `private` functions `postTest*WebResource` into one function taking one argument for the URLs with different protocols. Even further, you can also perform the `get` and `andExpect` checks in the same function.